File: VB/ChartsDemo.Wpf/ViewModels/OilPrices.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Xml.Serialization

Namespace ChartsDemo
    <XmlRoot("OilPrices")>
    Public Class OilPrices
        Inherits List(Of OilPricesByCompany)


        Private Shared dataSource_Renamed As List(Of OilPricesByCompany) = Nothing
        Public Shared ReadOnly Property DataSource() As List(Of OilPricesByCompany)
            Get
                If dataSource_Renamed Is Nothing Then
                    Dim s = New XmlSerializer(GetType(OilPrices))
                    dataSource_Renamed = DirectCast(s.Deserialize(DataLoader.LoadFromResources("/Data/OilPrices.xml")), List(Of OilPricesByCompany))
                End If
                Return dataSource_Renamed
            End Get
        End Property
        Public Shared ReadOnly Property EuropeBrentPrices() As List(Of OilPriceByDate)
            Get
                Return OilPrices.DataSource.First(Function(x) x.CompanyName = "Europe Brent").Prices
            End Get
        End Property
    End Class
    Public Class OilPricesByCompany
        Public Property CompanyName() As String
        Public Property Prices() As List(Of OilPriceByDate)
    End Class
    Public Class OilPriceByDate
        Public Property Timestamp() As Date
        Public Property MinValue() As Double
        Public Property MaxValue() As Double
    End Class
End Namespace

File: CS/ChartsDemo.Wpf/ViewModels/OilPrices.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Serialization;

namespace ChartsDemo {
    [XmlRoot("OilPrices")]
    public class OilPrices : List<OilPricesByCompany> {
        static List<OilPricesByCompany> dataSource = null;
        public static List<OilPricesByCompany> DataSource {
            get {
                if(dataSource == null) {
                    var s = new XmlSerializer(typeof(OilPrices));
                    dataSource = (List<OilPricesByCompany>)s.Deserialize(DataLoader.LoadFromResources("/Data/OilPrices.xml"));
                }
                return dataSource;
            }
        }
        public static List<OilPriceByDate> EuropeBrentPrices { get { return OilPrices.DataSource.First(x => x.CompanyName == "Europe Brent").Prices; } }
    }
    public class OilPricesByCompany {
        public string CompanyName { get; set; }
        public List<OilPriceByDate> Prices { get; set; }
    }
    public class OilPriceByDate {
        public DateTime Timestamp { get; set; }
        public double MinValue { get; set; }
        public double MaxValue { get; set; }
    }
}
